### ISSUE TITLE
🐞 Add optional tags param to listen_for_error

### ISSUE THREAD

**Issue by nnc** [appsignal/appsignal-ruby#353]
Would you please consider adding `tags` param to `listen_for_error`?
Seems to be a trivial addition, and would match the underlying `send_error` more closely.

We use `listen_for_error` in a few places and I wanted to add some tags to it, but then discovered that method currently takes none, unlike all other ways to manually send errors to AppSignal. Plus `tag_request` also doesn't help with `listen_for_error`, so it's a bit of unnecessary limitation IMO.

Granted, we could just hand roll this with `send_error`, but why not use the nicer `listen_for_error` when it's there already.

**Comment by tombruijn** [comment]
Hi @nnc ,

Thanks for the report!
Yes, this would be useful to add. We're thinking of revamping the `set_error`, `send_error`, `listen_for_error` and `monitor_transaction` helpers at some point, some ideas were posted in issue #176.

In the meantime I think it's a small change to support this for `listen_for_error`

Should become something like:

```ruby
    def listen_for_error(tags = nil, namespace = Appsignal::Transaction::HTTP_REQUEST)
      yield
    rescue Exception => error # rubocop:disable Lint/RescueException
      send_error(error, tags, namespace)
      raise error
    end
```

We'll try to squeeze this change in somewhere soon-ish

**Comment by nnc** [comment]
Thanks @tombruijn!

Looking forward to your proposed "soon-ish" solution. 😀

**Comment by tombruijn** [comment]
@nnc I created a PR that supports tags and namespaces for `Appsignal.listen_for_error`, see PR #357 . Can you try it out and let me know if it works like expected?

**Comment by tombruijn** [comment]
Fixed in #357. Let us know at support@appsignal.com when there's a problem with this change.
